File: homework7/AssignmentCode/GymAppCore/ClientApp/src/app/classes/Day.ts
import { Exercise } from './Exercise';

export class Day {
  name: string;
  dayId: number;
  workoutId: number;
  order: number;
  isRestDay: boolean;
  exercises: Exercise[];
  isDeleted: boolean;

    //Client Side Properties
    hasIncompleteSession: boolean;

  constructor(name: string, workoutId: number) {
    this.name = name;
    this.workoutId = workoutId;
    
    this.hasIncompleteSession = false;
  }
}

File: homework7/AssignmentCode/GymAppData/Models/Day.cs
namespace GymAppData.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    public class Day
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int DayId { get; set; }

        public int WorkoutId { get; set; }

        [Required]
        [StringLength(250)]
        public string Name { get; set; }

        public int Order { get; set; }

        public bool IsRestDay { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime Created { get; set; }

        public DateTime? Updated { get; set; }

        public bool IsDeleted { get; set; }

        public virtual Workout Workout { get; set; }

        public virtual List<Exercise> Exercises { get; set; }
    }
}
